Helper/Image.php

```diff
@@ -81,14 +81,20 @@ protected function initBaseFile()
      * Configurable::setImageFromChildProduct() only pulls 'image' type
      * and not the type set by the imageHelper
      *
-     * @return string
+     * @param \Magento\Catalog\Model\Product\Image $model
+     *
+     * @return mixed|string|null
      */
-    private function getProductImage()
+    private function getProductImage(\Magento\Catalog\Model\Product\Image $model)
     {
-        $imageUrl = $this->getProduct()->getImage();
-        if (!$this->getImageFile() && $this->getType() !== 'image'
-            && $this->getProduct()->getTypeId() == ProductTypeConfigurable::TYPE_CODE) {
-            $imageUrl = $this->getConfigurableProductImage() ?: $imageUrl;
+        if ($this->getImageFile()) {
```

Comment:
I don't really understand why you added this condition here as it's already checked on line70 and it seems to be the only place where the method getProductImage() is called (in the else statement)

Reply:
I must have been tired. I can cut it down to save some rows :D